File: FSharp/PP0604A.fs

let list = [1..1111]

let sum data = data |> List.sum

let avg (data:List<int>,func) = (func data) / data.Length
    
 

 
[<EntryPoint>]
let main argv =
    let result = avg(list,sum)
    printfn "%A" result
    0 // return an integer exit code

File: JS/PP0604A.js

(function() {
    var data = [1, 2, 3, 4, 5, 6, 4, 5];
    var sum = function (list) {
        return list.reduce(function(x, y) {
            return x + y;
        });
    }

    var main = function (list, func) {
        return func(list) / list.length;
    }

    console.log(main(data, sum));
})();

while (true) {
    
}
